sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py

```diff
@@ -500,14 +500,17 @@ def get_file_client(self, file  # type: Union[FileProperties, str]
         :param file:
             The file with which to interact. This can either be the name of the file,
             or an instance of FileProperties. eg. directory/subdirectory/file
-        :type file: str or ~azure.storage.filedatalake.FileProperties
+        :type file: str or ~azure.storage.filedatalake.FileProperties or a PurePosixPath
         :returns: A DataLakeFileClient.
         :rtype: ~azure.storage.filedatalake..DataLakeFileClient
         """
-        try:
-            file_path = file.name
-        except AttributeError:
-            file_path = self.path_name + '/' + file
```

Comment:
if we change it to this
```
        try:
            file_path = file.get('name')
        except AttributeError:
            file_path = self.path_name + '/' + str(file)
```
it will cover the case for PurePosixPath, so that we don't have to document it specifically

Reply:
That's a fantastic idea!